### PR TITLE
Upgrading Abseil to LTS 20200923.2

### PR DESCRIPTION
**Pull request by veblush**
Upgrading Abseil to LTS [20200923.2](https://github.com/abseil/abseil-cpp/releases/tag/20200923.2)

### CODE REVIEW

**Comment by veblush** [comment]
[grpc_build_artifacts_multiplatform](https://g3c.corp.google.com/results/invocations/cc427293-bcd6-4c02-b594-0d7b89ab3c50/targets): running

**Comment by veblush** [comment]
[grpc_basictests_objc_ios](https://source.cloud.google.com/results/invocations/1649b54f-0307-49ad-8f6c-2d677c77b797/targets): failed because of `ios-test-cfstream-tests`

```
Testing failed:
	Use of undeclared identifier '__tsan_mutex_read_lock'
	Use of undeclared identifier '__tsan_mutex_read_lock'
	Use of undeclared identifier '__tsan_mutex_try_lock'; did you mean 'pthread_mutex_trylock'?
** TEST FAILED **


The following build commands failed:
	CompileC /Users/kbuilder/Library/Developer/Xcode/DerivedData/CFStreamTests-eysagtowlpfcwyfevvouytnaykfx/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/abseil.build/Objects-normal-tsan/x86_64/mutex.o abseil/absl/synchronization/mutex.cc normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)
```

**Comment by veblush** [comment]
[grpc_build_artifacts_multiplatform](https://g3c.corp.google.com/results/invocations/ccec9ddd-c7c6-4917-8f2d-6d73fb520382): running

**Comment by veblush** [comment]
Failed PR tests:
- [Bazel C/C++ Dbg MacOS](https://source.cloud.google.com/results/invocations/2e208803-80f6-497a-aa06-2ea8381d5d39/targets) - //test/core/transport/chttp2:too_many_pings_test
- [Bazel RBE Opt C/C++](https://source.cloud.google.com/results/invocations/106c5101-51b3-45d3-a490-8cb4381f71fd/targets) - //test/core/end2end:h2_proxy_nosec_test@cancel_after_accept@poller=epoll1
- [Bazel RBE TSAN C/C++](https://source.cloud.google.com/results/invocations/92573855-d75f-4a83-b6e9-d5e0456f65ed/targets) - //test/cpp/end2end:xds_end2end_test@poller=epoll1

**Comment by veblush** [comment]
`//test/cpp/end2end:xds_end2end_test` seems to fail actually because of this PR. ([log](https://source.cloud.google.com/results/invocations/e725e948-1f08-4cdf-a0a8-1aba2354845c/targets))

File: tools/bazel.rc [hunk 1]
```diff
@@ -61,17 +58,17 @@ build:tsan --copt=-fsanitize=thread
 build:tsan --copt=-fno-omit-frame-pointer
 build:tsan --copt=-DGPR_NO_DIRECT_SYSCALLS
 build:tsan --copt=-DGRPC_TSAN
-# TODO(jtattermusch): ideally we would set --copt=-DTHREAD_SANITIZER (used by absl)
```
Comment (jtattermusch) [inline tools/bazel.rc:64]: why isn't this TODO applicable anymore?

**Review by jtattermusch — APPROVED** [review]
LGTM with a question.

File: tools/bazel.rc (hunk 1 above)
Comment (veblush) [reply tools/bazel.rc:64]: Abseil doesn't use `DTHREAD_SANITIZER` anymore ([link](https://github.com/abseil/abseil-cpp/blob/d1de75bf540f091b4dfc860713d556e578c0f158/absl/base/config.h#L690)). Instead, it starts to detect it using a feature.

**Pull request merged**